parlai/tasks/anli/agents.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+import json
+import os
+
+from parlai.core.teachers import DialogTeacher, MultiTaskTeacher
+from parlai.tasks.multinli.agents import (
+    convert_to_dialogData,
+    BICLASS_LABELS,
+    MULTINLI_LABELS,
+)
+
+from .build import build
+
+ANLI = 'ANLI'
+ANLI_PREFIX = 'anli_'
+ANLI_VERSION = 'v0.1'
+ANLI_LABEL_DICT = {'e': 'entailment', 'c': 'contradiction', 'n': 'neutral'}
+ANLI_PREMISE_KEY = 'context'
+ANLI_HYPO_KEY = 'hypothesis'
+ANLI_ANSWER_KEY = 'label'
+ANLI_ROUNDS = ['R1', 'R2', 'R3']
+
+
+def _path(opt):
+    build(opt)
+
+    dt = opt['datatype'].split(':')[0]
+
+    if dt == 'train':
+        suffix = 'train'
+    elif dt == 'valid':
+        suffix = 'dev'
+    elif dt == 'test':
+        suffix = 'test'
+    else:
+        raise RuntimeError('Not valid datatype.')
+
+    data_path = os.path.join(
+        opt['datapath'],
+        ANLI,
+        ANLI_PREFIX + ANLI_VERSION,
+        opt['round'],
+        suffix + '.jsonl',
+    )
+
+    return data_path
+
+
+class RoundBaseTeacher(DialogTeacher):
+    @staticmethod
+    def add_cmdline_args(parser):
+        parser = parser.add_argument_group('RoundBase Teacher Args')
+        parser.add_argument(
+            '--dialog-format',
+            type='bool',
+            default=False,
+            help="True if one would like to convert to a dialogue format without special tokens such as 'Premise'"
+            " and 'Hypothesis' (default: False).",
+        )
+
+    def __init__(self, opt, shared=None):
+        opt = copy.deepcopy(opt)
+        opt['round'] = (
```

Comment:
still need to change this part, right? let's not mess with `opt` inside the init, but rather set `self.round`.